plugin/health_check.go

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package plugin
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/mattermost/mattermost-server/mlog"
+)
+
+const (
+	PLUGIN_HEALTH_CHECK_INTERVAL  = 30 // seconds
+	HEALTH_CHECK_PING_FAIL_LIMIT  = 3
+	HEALTH_CHECK_RESTART_LIMIT    = 3
+	HEALTH_CHECK_DISABLE_DURATION = 60 // minutes
```

Comment:
Here's a description for each variable here:

PLUGIN_HEALTH_CHECK_INTERVAL - How often we start the sequential check of all plugins

HEALTH_CHECK_PING_FAIL_LIMIT - How many failed calls to `client.Ping()` in a given plugin's health check before we deem the RPC connection to be unhealthy

HEALTH_CHECK_RESTART_LIMIT - How many times we restart the plugin before permanently deactivating it

HEALTH_CHECK_DISABLE_DURATION - How long we allow successive failures to occur before we permanently deactivate the plugin